google-cloud-spanner/src/main/java/com/google/cloud/spanner/StructReader.java

```diff
@@ -134,6 +138,56 @@ default String getPgJsonb(String columnName) {
     throw new UnsupportedOperationException("method should be overwritten");
   }
 
+  /**
+   * To get the proto message of generic type {@code T} from Struct.
+   *
+   * @param columnIndex Index of the column.
+   * @param m Proto message object. @see <a
+   *     href="https://developers.google.com/protocol-buffers/docs/reference/java-generated#message">getDefaultInstance()</a>
+   * @return The value of a non-{@code NULL} column with type {@link Type#proto(String)} ()}.
+   */
+  default <T extends AbstractMessage> T getProtoMessage(int columnIndex, T m) {
+    throw new UnsupportedOperationException("method should be overwritten");
+  }
+
+  /**
+   * To get the proto message of type {@code T} from Struct.
+   *
+   * @param columnName Name of the column.
+   * @param m Proto message object. @see <a
+   *     href="https://developers.google.com/protocol-buffers/docs/reference/java-generated#message">getDefaultInstance()</a>
+   * @return The value of a non-{@code NULL} column with type {@link Type#proto(String)} ()}.
+   */
+  default <T extends AbstractMessage> T getProtoMessage(String columnName, T m) {
+    throw new UnsupportedOperationException("method should be overwritten");
+  }
+
+  /**
+   * To get the proto enum of type {@code T} from Struct.
+   *
+   * @param columnIndex Index of the column.
+   * @param method Lambda method which takes a primitive int and return Enum. @see <a
+   *     href="https://developers.google.com/protocol-buffers/docs/reference/java-generated#enum">forNumber</a>
```

Comment:
This needs some more explanation on what the developer should do. Why do they need to provide a function that does this conversion?
Also (nit): The function that is provided does not need to be specified as a lambda expression, so I think the comment should read something like 'A function that takes ...'